great_expectations/data_context/store/datasource_store.py

```diff
@@ -218,3 +222,12 @@ def _determine_datasource_key(self, datasource_name: str) -> DataContextVariable
             resource_name=datasource_name,
         )
         return datasource_key
+
+    def _validate_key(self, key: Union[GeCloudIdentifier, Tuple[str, ...]]) -> None:  # type: ignore[override]
```

Comment:
This is a great opportunity to generalize Key into an object that is self-validating, as opposed to making every consumer validate the key using what is effectively a utility method

Reply:
Totally agree. I think we need to be careful to be backward compatible though since we currently use tuples as keys. Could this be split into a separate JIRA item?

Reply:
This method is removed, and ticket great-1241 created to generalize and add self-validation.